### PR TITLE
use _MSVC_LANG to include C++17 headers in Visual Studio

### PR DESCRIPTION
**Pull request by marciso**
There are also some fixes to the Windows build

### CODE REVIEW

**Comment by marciso** [comment]
Note that hunter needs to be upgraded to compile with MSVC 2019.

File: hdr/sqlite_modern_cpp/utility/utf16_utf8.h [hunk 1]
```diff
@@ -17,7 +17,7 @@ namespace sqlite {
 			std::size_t produced_output = 0;
 			while(true) {
 				char *used_output;
-				switch(codecvt.out(state, remaining_input, &input[input.size()],
```
Comment (marciso) [inline hdr/sqlite_modern_cpp/utility/utf16_utf8.h]: fixing UB

File: hdr/sqlite_modern_cpp.h [hunk 2]
```diff
@@ -29,7 +29,7 @@ namespace sqlite {
 	template<class T, bool Name = false>
 	struct index_binding_helper {
 		index_binding_helper(const index_binding_helper &) = delete;
-#if __cplusplus < 201703
+#if __cplusplus < 201703 || _MSVC_LANG <= 201703
```
Comment (marciso) [inline hdr/sqlite_modern_cpp.h]: Note, MSVC needs the move constructor to be explicit

**Comment by aminroosta** [comment]
Thanks @marciso

**Pull request merged**